src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2885,8 +2903,8 @@ private BoundExpression BindArrayCreationExpression(ArrayCreationExpressionSynta
                 {
                     if (arg.Kind() != SyntaxKind.OmittedArraySizeExpression)
                     {
-                        var size = BindValue(arg, diagnostics, BindValueKind.RValue);
-                        Error(diagnostics, ErrorCode.ERR_InvalidArray, dimension[0]);
+                        var size = BindRValueWithoutTargetType(arg, diagnostics);
+                        Error(diagnostics, ErrorCode.ERR_InvalidArray, arg);
```

Comment:
>arg [](start = 71, length = 3)

Why change the error location?

Reply:
Because the problem is with this arg, not `dimensions[0]`.  It was simply wrong before.  Moreover, `dimensions[0]` might be an omitted size expression, which has no width.

---
In reply to: [301336868](https://github.com/dotnet/roslyn/pull/37052#discussion_r301336868) [](ancestors = 301336868)